### PR TITLE
Tutorial updated to newer APIs and the testing examples have been expanded a bit #2334

### PR DESCRIPTION
**Pull request by rohdef**
https://github.com/http4s/http4s/issues/2334 and https://github.com/http4s/http4s/issues/1501

### CODE REVIEW

File: docs/src/main/tut/dsl.md [hunk 1]
```diff
@@ -86,7 +86,17 @@ But here in the REPL, it's up to us to run it:
...
+Cool.  Now we will be able to test things like the status:
+
+```tut
+val status = response.status
+```
+
+Or the content of our body:
+
+```tut
+val body = response.bodyAsText.compile.toList.unsafeRunSync().mkString("")
```
Comment (rossabaker) [inline docs/src/main/tut/dsl.md]: How about just `response.as[String].unsafeRunSync()`?

File: docs/src/main/tut/dsl.md [hunk 2]
```diff
@@ -68,7 +68,7 @@ and experiment directly in the REPL.
 ```tut
 val getRoot = Request[IO](Method.GET, uri("/"))
 
-val io = service.orNotFound.run(getRoot)
+val io = service.apply(getRoot).getOrElseF(NotFound())
```
Comment (rossabaker) [inline docs/src/main/tut/dsl.md]: I think this was explained to me elsewhere, but I can't find the reference now.  What does the existing code do that's wrong?  I would think these are equivalent.

Comment (rohdef) [reply docs/src/main/tut/dsl.md]: At least in my case it seems to type check but once I run it, it cannot find orNotFound :/

File: docs/src/main/tut/dsl.md (hunk 1 above)
Comment (rohdef) [reply docs/src/main/tut/dsl.md]: Will try that out, that looks a lot better imo

Comment (rohdef) [reply docs/src/main/tut/dsl.md]: And now that I try it again, I naturally cannot reproduce... strange, wonder if I did a typo or if I hit some kind of corner case :S

File: docs/src/main/tut/dsl.md (hunk 2 above)
Comment (rohdef) [reply docs/src/main/tut/dsl.md]: And now that I try it again, I naturally cannot reproduce... strange, wonder if I did a typo or if I hit some kind of corner case :S

Comment (rossabaker) [reply docs/src/main/tut/dsl.md]: There needs to be an import.  In master, it's `org.http4s.implicits._`.  You might have been missing the import, or I think it also requires -Ypartial-unification.

**Comment by armanbilge** [comment]
Both issues are closed, and 0.18.x is history.

**Pull request closed without merging**